File: 0551-student-attendance-record-i/0551-student-attendance-record-i.cpp
class Solution {
public:
    bool checkRecord(string s) {
        int absences = 0, lates = 0;
        for (char chr : s) {
            switch(chr) {
                case 'A':
                    absences++;
                    if (absences == 2) return false;
                    lates = 0;
                    break;
                case 'L':
                    lates++;
                    if (lates == 3) return false;
                    break;
                case 'P':
                    lates = 0;
                    break;
            }
        }
        
        return true;
    }
};

File: 0551-student-attendance-record-i/0551-student-attendance-record-i.rb
# @param {String} s
# @return {Boolean}
def check_record(s)
    absences = consec_late = 0
    s.chars.each_index do |i|
        case s[i]
        when "A"
            absences+=1
        when "L"
            s[i-1] == 'L' ? consec_late += 1 : consec_late = 1
        end
    end
    absences < 2 && consec_late < 3 ? true : false

end